include/GL/glui.h

```diff
@@ -460,6 +476,43 @@ class GLUIAPI GLUI_StdBitmaps
 
 /************************************************************/
 /*                                                          */
+/*                   Copy/Paste GLUI Class                  */
+/*                                                          */
+/************************************************************/
+
+class GLUIAPI GLUI_Clipboard_Object 
+{
+public:
+
+	GLUI_Clipboard_Object(const char *name);
+
+	/**
+	 * Typically @c name is CLIPBOARD or PRIMARY.
+	 * PRIMARY is available only on X11 systems, and
+	 * should be set whenever a selection is finalized.
+	 */
+	const GLUI_String name;
+	
+	/**
+	 * This is analogous to copying to the clipboard.
+	 */
+	void set_text(const char *sel, size_t len);	
+
+	/**
+	 * This is analogous to pasting from the clipboard, or
+	 * that is, getting the text on the clipboard to paste.
+	 */
+	bool get_text(GLUI_String &container);
+};
+
+/**
+ * These are built-in clipboards. On non X11-like systems
+ * @c GLUI_Selection is a "no-op" object.
+ */
+extern GLUI_Clipboard_Object GLUI_Clipboard, GLUI_Selection;
```

Comment:
Yes, it will work.  In an object oriented scheme these statics would ordinarily be wrapped in a namespace or a class, but aside from that it seems unremarkable to me.